### PR TITLE
Bug 1484739 - use taskcluster-lib-pulse

### PR DESCRIPTION
**Pull request by djmitche**
This removes the need for pulse credentials in `user-config.yml`.  It also fails gracefully if `pulse.namespace` isn't set -- you just won't get any subscriptions.

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -22,10 +23,9 @@ defaults:
 
   # Pulse credentials
   pulse:
-    username: !env PULSE_USERNAME
-    password: !env PULSE_PASSWORD
-    hostname: !env PULSE_HOSTNAME
-    vhost: !env PULSE_VHOST
```
[the remark this answers is not in the document]
Comment (djmitche) [reply config.yml]: Indeed I should!

**Comment by djmitche** [comment]
OK, I think I fixed both issues.

File: src/PulseEngine/index.js [hunk 2]
```diff
@@ -1,71 +1,217 @@
...
+      await Promise.all(
+        subscriptions.map(({ pattern, exchange }) =>
+          channel.bindQueue(queueName, exchange, pattern)
+        )
+      );
+
+      const { consumerTag } = await channel.consume(queueName, message => {
+        const payload = JSON.parse(message.content.toString('utf8'));
+
+        onMessage({ [eventName]: payload });
```
[the remark this answers is not in the document]
Comment (djmitche) [reply src/PulseEngine/index.js]: No, you have to bind a queue before you can consume from it :)

[the remark this answers is not in the document]
Comment (djmitche) [reply src/PulseEngine/index.js]: and assert it before you can bind to it

[the remark this answers is not in the document]
Comment (helfi92) [reply src/PulseEngine/index.js]: Makes sense.

**Comment by djmitche** [comment]
lint'd :)

**Pull request merged**